### PR TITLE
[examples] update autoparallel tutorial demo

### PR DESCRIPTION
**Pull request by YuliangLiu0306**
This PR updates the auto-parallel tutorial demo with the lastest api.

### CODE REVIEW

**Comment by FrankLeeeee** [comment]
Can you add a `test_ci.sh` for this auto parallel example?

File: examples/tutorial/auto_parallel/test_ci.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -euxo pipefail
+
+pip install titans
+pip install pulp
+conda install coin-or-cbc
```
Comment (FrankLeeeee) [inline examples/tutorial/auto_parallel/test_ci.sh:6]: `conda install -c conda-forge coin-or-cbc`

File: examples/tutorial/auto_parallel/test_ci.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+colossalai run --nproc_per_node 4 auto_parallel_with_resnet.py -s
```
Comment (FrankLeeeee) [inline examples/tutorial/auto_parallel/test_ci.sh:7]: `-s` is no longer needed if we only use synthetic dataset. This tutorial is only for user to quickly start a script without any preparation.

File: examples/tutorial/auto_parallel/test_ci.sh (hunk 1 above)
Comment (FrankLeeeee) [inline examples/tutorial/auto_parallel/test_ci.sh:6]: Make it into a conda `environment.yaml` and include pytorch and colossalai.

Comment (YuliangLiu0306) [reply examples/tutorial/auto_parallel/test_ci.sh:6]: Okay

Comment (YuliangLiu0306) [reply examples/tutorial/auto_parallel/test_ci.sh:6]: Ok

File: examples/tutorial/auto_parallel/test_ci.sh (hunk 2 above)
Comment (YuliangLiu0306) [reply examples/tutorial/auto_parallel/test_ci.sh:7]: Sure

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**